xformers/ops/fmha/triton_splitk.py

```diff
@@ -1028,19 +1028,31 @@ def not_supported_reasons(cls, d: Inputs) -> List[str]:
     @classmethod
     def get_split_k(cls, B: int, G: int, H: int, Mk: int) -> int:
         """Heuristic for the number of splits"""
+        print(f"B = {B}, G = {G}, H = {H}, Mk = {Mk}")
         bh = max(B * H, 1)  # NOTE: Handle B*h=0 case
-        split_k = max(Mk, 1024) // bh
+        split_k = max(Mk + bh - 1, 1024) // bh
         if torch.version.hip:
             max_chunk_size = 64
-            split_k_stop_val = min(Mk / max_chunk_size, 1024 / (B * G * H))
+            split_k_stop_val = 1024 / (B * G * H)
+            while split_k > 0 and Mk / (split_k - 1) < max_chunk_size:
+                split_k = split_k - 1
+
+            while split_k > split_k_stop_val:
+                split_k = split_k // 2
+
+            split_size = (Mk + split_k - 1) // split_k
+            chunk_size = split_size // max_chunk_size * max_chunk_size
+            if chunk_size < split_size:
+                split_k += 1
+
             split_k_upper_bound = 512
         else:
             max_chunk_size = 64 if Mk <= 512 and bh <= 64 else 128
             split_k_stop_val = Mk / max_chunk_size
             split_k_upper_bound = 64
 
-        while split_k > split_k_stop_val:
-            split_k = split_k // 2
+            while split_k > split_k_stop_val:
```

Comment:
Confirming if that this is an intended change ?

Reply:
Sorry, it seems like I pushed an unexpected commit to this PR. I just convert this PR to draft. will let you know when I cleaned up the code.

Reply:
Just double checked it, we try to make the changes only impact the hip side, so make the above changes here.